components/feature/downloads/src/test/java/mozilla/components/feature/downloads/DownloadCancelDialogFragmentTest.kt

```diff
@@ -0,0 +1,135 @@
+package mozilla.components.feature.downloads
+
+import android.graphics.drawable.GradientDrawable
+import android.view.Gravity
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentTransaction
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.feature.downloads.ui.DownloadCancelDialogFragment
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.doReturn
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+@Config(application = TestApplication::class)
+class DownloadCancelDialogFragmentTest {
+
+    @Test
+    fun `WHEN accept button is clicked THEN onAcceptClicked must be called`() {
+        spy(DownloadCancelDialogFragment.newInstance(2)).apply {
+            doReturn(testContext).`when`(this).requireContext()
+            doReturn(mockFragmentManager()).`when`(this).parentFragmentManager
+            with(onCreateDialog(null)) {
+                findViewById<Button>(R.id.accept_button).apply { performClick() }
+            }
+            verify(this).onAcceptClicked
+        }
+    }
+
+    @Test
+    fun `WHEN deny button is clicked THEN onDenyClicked must be called`() {
+        spy(DownloadCancelDialogFragment.newInstance(2)).apply {
+            doReturn(testContext).`when`(this).requireContext()
+            doReturn(mockFragmentManager()).`when`(this).parentFragmentManager
+            with(onCreateDialog(null)) {
+                findViewById<Button>(R.id.deny_button).apply { performClick() }
+            }
+            verify(this).onDenyClicked
+        }
+    }
+
+    @Test
+    fun `dialog MUST adhere to overriding strings`() {
```

Comment:
nit: For new tests we try to follow the [GIVEN, THEN, WHEN naming pattern](https://martinfowler.com/bliki/GivenWhenThen.html)